netbox/utilities/views.py

```diff
@@ -39,6 +41,22 @@ def dispatch(self, request, *args, **kwargs):
         return super().dispatch(request, *args, **kwargs)
 
 
+class TokenConditionalLoginRequiredMixin(ConditionalLoginRequiredMixin):
+    def dispatch(self, request, *args, **kwargs):
+        # Attempt to authenticate the user using a DRF token, if provided
+        if settings.LOGIN_REQUIRED and not request.user.is_authenticated:
+            authenticator = TokenAuthentication()
+            auth_info = authenticator.authenticate(request)
+            if auth_info is not None:
+                request.user = auth_info[0]  # User object
+                try:
+                    request.auth = auth_info[1]
+                except IndexError:
+                    pass
```

Comment:
If `auth_info` is not None, I think it should be safe to assume that the second item will be there. Granted we've over-ridden that method, but we shouldn't ever be changing the return signature.

But, now I'm beginning to wonder if you ran in to something unexpected that led to this ...

Reply:
Did a double check, I think I did this out of habit.  Checking it, it should include the token information so like you said, it should be safe.